src/List/ListItem.tsx

```diff
@@ -1,18 +1,16 @@
-import * as React from 'react';
+import React from 'react';
 import { StyledListItem, StyleProps } from './ListItem.styles';
 
-export interface Props extends StyleProps {
-  children: React.ReactNode;
-}
-
-export const ListItem = (props: Props): JSX.Element => {
-  const { disabled, children, ...rest } = props;
+export const ListItem: React.FC<StyleProps> = props => {
+  const { disabled, children, onClick, ...rest } = props;
```

Comment:
I know this isn't from now, but could these props not be destructured on the line above?